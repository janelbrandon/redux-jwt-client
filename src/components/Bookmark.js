import React from 'react'
import PropTypes from 'prop-types'
import { Button } from '../styles/componentSyles'


function Bookmark(props) {
  const { _id, title, url, remove } = props
  return (
    <li>
      {title} (<a href={url}>Visit</a>)
      <Button onClick={() => remove(_id)}>Delete!</Button>
    </li>
  )
}

Bookmark.propTypes = {
  _id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  remove: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired
}

export default Bookmark
