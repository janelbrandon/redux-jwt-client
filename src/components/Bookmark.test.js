import React from 'react'
import renderer from 'react-test-renderer'
import Bookmark from './Bookmark'
import { shallow } from 'enzyme'

jest.mock('../services/BookmarkService')
let { removeBookmark } = require('../services/BookmarkService')

describe('Bookmark snapshot', () => {
    it('Should render as expected', () => {
        const tree = renderer.create(
            <Bookmark key='123' title='My Bookmark' url="http://www.mybookmark.com" _id="123" remove={() => { }} />
        )
        expect(tree.toJSON()).toMatchSnapshot()
    })
})

describe('Bookmark component', () => {
    it('should call remove with correct argument when delete button is clicked', () => {
        const _id = "123"

        const removeSpy = jest.fn((id) => {
            console.log('in removeSpy with id', id)
        })

        const wrapper = shallow(
            <Bookmark _id={_id} title="My fake bookmark" url="http://www.myfakebookmark" remove={removeSpy} />
        )
        let btn = wrapper.find('button')
        expect(btn).toHaveLength(1)
        expect(btn.text()).toEqual('Delete!')
        btn.prop('onClick')()
        expect(removeSpy).toBeCalledWith(_id)
    })
    it('should call removeBookmark in BookmarkService', () => {
        const _id = "123"
        const wrapper = shallow(
            <Bookmark _id={_id} title="My fake bookmark" url="http://www.myfakebookmark" remove={removeBookmark} />
        )

        let btn = wrapper.find('button')
        expect(btn).toHaveLength(1)
        expect(btn.text()).toEqual('Delete!')
        btn.prop('onClick')()
    })


})