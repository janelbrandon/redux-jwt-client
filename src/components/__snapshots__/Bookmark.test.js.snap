// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark snapshot Should render as expected 1`] = `
<li>
  My Bookmark
   (
  <a
    href="http://www.mybookmark.com"
  >
    Visit
  </a>
  )
  <button
    className="sc-bdVaJa xKgSf"
    onClick={[Function]}
  >
    Delete!
  </button>
</li>
`;
